aggregator_core/src/datastore.rs

```diff
@@ -4958,6 +4967,61 @@ DELETE FROM global_hpke_keys WHERE config_id = $1",
         )
     }
 
+    #[tracing::instrument(skip(self), err(level = Level::DEBUG))]
+    pub async fn rewrite_global_hpke_keypairs(
+        &self,
+        keypairs: &[GlobalHpkeKeypair],
+    ) -> Result<(), Error> {
+        let stmt = self
+            .prepare_cached(
+                "-- rewrite_global_hpke_keypairs()
+DELETE FROM global_hpke_keys",
+            )
+            .await?;
+        self.execute(&stmt, &[]).await?;
+
+        try_join_all(keypairs.iter().map(|keypair| async move {
+            let hpke_config_id = u8::from(*keypair.id()) as i16;
+            let hpke_config = keypair.hpke_keypair().config().get_encoded()?;
+            let encrypted_hpke_private_key = self.crypter.encrypt(
+                "global_hpke_keys",
+                &u8::from(*keypair.id()).to_be_bytes(),
+                "private_key",
+                keypair.hpke_keypair().private_key().as_ref(),
+            )?;
+
+            let stmt = self
+                .prepare_cached(
+                    "-- rewrite_global_hpke_keypair()
+INSERT INTO global_hpke_keys
+    (config_id, config, private_key, state, last_state_change_at, created_at, updated_at, updated_by)
+    VALUES ($1, $2, $3, $4, $5, $6, $7, $8)",
+                )
+                .await?;
+            let now = self.clock.now().as_naive_date_time()?;
+            check_insert(
+                self.execute(
+                    &stmt,
+                    &[
+                        /* config_id */ &hpke_config_id,
+                        /* config */ &hpke_config,
+                        /* private_key */ &encrypted_hpke_private_key,
+                        /* state */ keypair.state(),
+                        /* last_state_change_at */
+                        &keypair.last_state_change_at().as_naive_date_time()?,
+                        /* created_at */ &now,
```

Comment:
Good point.

I went with changing `rewrite_global_hpke_keypairs` to upserting rows, so that `created_at` touches as little application code as possible.